syntax = "proto3";

package filemetadata;

message FileMetadata {
    string file_name = 1;
    int64 file_size = 2;  
    string file_type = 3; 
}

message Empty {}

message GetMetadataRequest {
    string file_name = 1;
}

message MetadataList {
    repeated FileMetadata files = 1;
}

service FileMetadataService {
    rpc UploadMetadata (FileMetadata) returns (Empty);
    rpc ListMetadata (Empty) returns (stream FileMetadata);
    rpc GetMetadataByName (GetMetadataRequest) returns (FileMetadata);
}